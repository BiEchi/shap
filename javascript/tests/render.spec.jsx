import renderer from 'react-test-renderer';
import { AdditiveForceVisualizer } from "../visualizers";
import React from 'react';

it('renders correctly', () => {
  const tree = renderer
    .create(<AdditiveForceVisualizer
      baseValue={0.0}
      link={"identity"}
      featureNames={{
        "0": "Blue",
        "1": "Red",
        "2": "Green",
        "3": "Orange"
      }}
      outNames={["color rating"]}
      features={{
        "0": { value: 1.0, effect: 1.0 },
        "1": { value: 0.0, effect: 0.5 },
        "2": { value: 2.0, effect: -2.5 },
        "3": { value: 2.0, effect: -0.5 }
      }}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
