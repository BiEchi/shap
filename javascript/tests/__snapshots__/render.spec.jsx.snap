// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<svg
  style={
    {
      "display": "block",
      "fontFamily": "arial",
      "sansSerif": true,
      "userSelect": "none",
    }
  }
>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "
          .force-bar-axis path {
            fill: none;
            opacity: 0.4;
          }
          .force-bar-axis paths {
            display: none;
          }
          .tick line {
            stroke: #000;
            stroke-width: 1px;
            opacity: 0.4;
          }
          .tick text {
            fill: #000;
            opacity: 0.5;
            font-size: 12px;
            padding: 0px;
          }",
      }
    }
  />
</svg>
`;
